### ISSUE TITLE
Plugins are not updated

### ISSUE THREAD

**Issue by petrushenko** [willsoto/ng-chartist#545]
I'm using plugin [_'chartist-plugin-legend'_](https://github.com/CodeYellowBV/chartist-plugin-legend) and I'm trying to update chart data and options (with plugins), and data and other options are updated but the plugin is not. How can I update plugins too?
With [_'chartist-plugin-axistitle'_]('chartist-plugin-axistitle') the same thing

**Comment by petrushenko** [comment]
Here is an example _'with chartist-plugin-axistitle'_ :
https://stackblitz.com/edit/angular-ivy-umsaa9?file=src/app/app.component.ts
I am updating the graph and save a new label for the axis title, but it is not updated.
Maybe this is the problem with these plugins?

**Comment by hakimio** [comment]
Most likely it's a problem with plugins.
Anyway, you can use `initialized` ng-chartist event to get chartist instance and then you can manually try to call `update()` to see if it makes any difference: `this.chart.update(data, options)`. If it doesn't, those plugins require the whole chart to be re-rendered to apply config changes.

**Comment by petrushenko** [comment]
> Most likely it's a problem with plugins.
> Anyway, you can use `initialized` ng-chartist event to get chartist instance and then you can manually try to call `update()` to see if it makes any difference: `chart.update(data, options)`. If it doesn't, those plugins require the whole chart to be re-rendered to apply config changes.
> I don't see anything wrong with your code.

Yes, you right, those plugins require re-rendering: `chart.update()` did not help, and I tried to disable and enable the chart component and it worked, are there any options for re-rendering the chart from code?

**Comment by hakimio** [comment]
Theoretically you could manually destroy the chart, to force it to be recreated on next update (haven't tested though):
```ts
ngChartistCmp['chart'].detach();
ngChartistCmp['chart'] = null;

this.options = {/* ... */};
```

**Comment by petrushenko** [comment]
@hakimio thank you, this code refresh my component:
```
    rerenderChartist(): void {
        if (!this.chartist){
            return;
        }

        this.chartist?.ngOnDestroy();
        this.chartist?.ngOnInit();
    }
````
Solved!
